rllib/core/testing/tf/bc_rl_trainer.py

```diff
@@ -3,28 +3,17 @@
 
 from ray.rllib.core.rl_trainer.tf.tf_rl_trainer import TfRLTrainer
 from ray.rllib.policy.sample_batch import SampleBatch
-from ray.rllib.policy.sample_batch import MultiAgentBatch
+
 from ray.rllib.core.testing.testing_trainer import BaseTestingTrainer
+from ray.rllib.utils.typing import TensorType
 
 
 class BCTfRLTrainer(TfRLTrainer, BaseTestingTrainer):
-    def compute_loss(
-        self, fwd_out: MultiAgentBatch, batch: MultiAgentBatch
+    def _compute_loss_per_module(
+        self, module_id: str, batch: SampleBatch, fwd_out: Mapping[str, TensorType]
     ) -> Mapping[str, Any]:
 
-        loss_dict = {}
-        loss_total = None
-        for module_id in fwd_out:
-            action_dist = fwd_out[module_id]["action_dist"]
-            loss = -tf.math.reduce_mean(
-                action_dist.log_prob(batch[module_id][SampleBatch.ACTIONS])
-            )
-            loss_dict[module_id] = loss
-            if loss_total is None:
-                loss_total = loss
-            else:
-                loss_total += loss
-
-        loss_dict[self.TOTAL_LOSS_KEY] = loss_total
+        action_dist = fwd_out["action_dist"]
+        loss = -tf.math.reduce_mean(action_dist.log_prob(batch[SampleBatch.ACTIONS]))
 
-        return loss_dict
+        return {self.TOTAL_LOSS_KEY: loss}
```

Comment:
maybe this should get saved under per-module key, so that rl_trainer can always aggregate the loss into a total key.
also, why is TOTAL_LOSS_KEY a contains on self?

Reply:
to be accessible everywhere. No need to import :)

Reply:
> maybe this should get saved under per-module key, so that rl_trainer can always aggregate the loss into a total key.

This is more general. Trust me :)

Reply:
ok :)